src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -161,12 +186,48 @@ private void AppendPath(ODataPath path)
 
                 case ODataPathKind.Operation:
                 case ODataPathKind.OperationImport:
+                    if (kind == ODataPathKind.Operation)
+                    {
+                        foreach (var kvp in _dollarCountPaths)
+                        {
+                            ODataPath dollarCountPath = kvp.Value.FirstOrDefault(p => DollarCountAndOperationPathsSimilar(p, path));
+                            if (dollarCountPath != null &&
+                                _allNavigationSourcePaths.TryGetValue(kvp.Key, out IList<ODataPath> dollarPathList))
+                            {
+                                dollarPathList.Remove(dollarCountPath);
+                                break;
+                            }
+                        }
+                    }
+
                     _allOperationPaths.Add(path);
                     break;
 
                 default:
                     return;
             }
+
+            bool DollarCountAndOperationPathsSimilar(ODataPath path1, ODataPath path2)
+            {
+                string countIdentifier = "count";
+
+                if ((path1.Kind == ODataPathKind.DollarCount && 
+                    path2.Kind == ODataPathKind.Operation && path2.LastSegment.Identifier.Equals(countIdentifier, StringComparison.OrdinalIgnoreCase)) ||
+                    (path2.Kind == ODataPathKind.DollarCount &&
+                    path1.Kind == ODataPathKind.Operation && path1.LastSegment.Identifier.Equals(countIdentifier, StringComparison.OrdinalIgnoreCase)))
+                {
+                    return GetModifiedPathItemName(path1).Equals(GetModifiedPathItemName(path2), StringComparison.OrdinalIgnoreCase);
+                }
+
+                return false;                
+            }
+
+            string GetModifiedPathItemName(ODataPath path)
+            {
+                IEnumerable<ODataSegment> modifiedSegments = path.Take(path.Count - 1);
```

Comment:
return the original if ! any()

Reply:
The suggestion is unclear. Which original if ! any()?

Reply:
```suggestion
                if (!path.Any()) return path;
                IEnumerable<ODataSegment> modifiedSegments = path.Take(path.Count - 1);
```

Reply:
Went with `if (!path.Any()) return null;`